modules/autotagging-commons/common/src/main/java/org/opensearch/rule/storage/AttributeValueStore.java

```diff
@@ -40,11 +39,21 @@ default void remove(K key, V value) {
     void remove(K key);
 
     /**
-     * Returns the values associated with the key
-     * @param key in the data structure
+     * Returns the values associated with the given key, including prefix matches,
+     * along with their match scores.
+     * For example, searching for "str" may also return results for "st" and "s".
+     * @param key the key to look up
+     */
+    default Map<V, Float> getWithMatchScore(K key) {
```

Comment:
Since score is a information hidden in `Match` we don't need to include the score in the method name as that is redundant and confusing.